service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -0,0 +1,219 @@
+    private static final String DEFAULT_QUANTITY_VALUE = "1";
+    private static final String DEFAULT_QUANTITY_TEXT = "Unk UoM";
+    private static final List<String> ACUTE_PRESCRIPTION_TYPE_CODES = Arrays.asList("acute", "acute-handwritten");
+    private static final List<String> REPEAT_PRESCRIPTION_TYPE_CODES =
+        Arrays.asList("delayed-prescribing", "repeat", "repeat-dispensing");
+    private static final String ACUTE_REPEAT_VALUE = "0";
+
+    private final MessageContext messageContext;
+    private final CodeableConceptCdMapper codeableConceptCdMapper;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    public String mapMedicationRequestToMedicationStatement(MedicationRequest medicationRequest) {
+        var medicationStatementTemplateParameters = MedicationStatementTemplateParameters.builder()
+            .medicationStatementId(messageContext.getIdMapper().getOrNew(ResourceType.MedicationRequest, medicationRequest.getId()))
+            .statusCode(buildStatusCode(medicationRequest))
+            .effectiveTime(StatementTimeMappingUtils.prepareEffectiveTimeForMedicationRequest(medicationRequest))
+            .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForMedicationRequest(medicationRequest))
+            .medicationReferenceCode(buildMedicationReferenceCode(medicationRequest))
+            .ehrSupplyId(messageContext.getMedicationRequestIdMapper().getOrNew(medicationRequest.getId()))
+            .medicationStatementPertinentInformation(buildDosageInstructionPertinentInformation(medicationRequest))
+            .ehrSupplyPertinentInformation(buildPertinentInformation(medicationRequest))
+            .repeatNumber(buildRepeatValue(medicationRequest))
+            .quantityValue(buildQuantityValue(medicationRequest))
+            .quantityText(buildQuantityText(medicationRequest))
+            .ehrSupplyDiscontinueCode(buildStatusReasonCode(medicationRequest))
+            .ehrSupplyDiscontinueId(randomIdGeneratorService.createNewId())
+            .ehrSupplyDiscontinueAvailabilityTime(buildStatusReasonAvailabilityTime(medicationRequest))
+            .priorPrescriptionId(buildPriorPrescription(medicationRequest))
+            .basedOn(buildBasedOn(medicationRequest))
+            .build();
+
+        if (medicationRequest.getIntent().getDisplay().equals(MedicationRequestIntent.PLAN.getDisplay())) {
+            return TemplateUtils.fillTemplate(MEDICATION_STATEMENT_AUTHORISE_TEMPLATE, medicationStatementTemplateParameters);
+        } else if (medicationRequest.getIntent().getDisplay().equals(MedicationRequestIntent.ORDER.getDisplay())) {
+            return TemplateUtils.fillTemplate(MEDICATION_STATEMENT_PRESCRIBE_TEMPLATE, medicationStatementTemplateParameters);
+        }
+
+        throw new EhrMapperException("Could not map Medication Request intent");
+    }
+
+    private String buildStatusCode(MedicationRequest medicationRequest) {
```

Comment:
Add a check for status code being present as it's 1..1 and if not present will continue down your complete route.